File: solutions/go/flatten-array/3/flatten_array.go
package flatten

func Flatten(nested any) []any {
    result := make([]any, 0, 1)
	switch value := nested.(type) {
    case []any:
        for _, elem := range value {
            result = append(result, Flatten(elem)...)
        }
    case any:
        result = append(result, value)
    default:
        return nil
    }
    return result
}

File: solutions/lua/flatten-array/1/flatten-array.lua
local function is_array(tbl)
  if type(tbl) ~= 'table' then
    return false
  end
  local i = 0
  for _ in pairs(tbl) do
      i = i + 1
      if tbl[i] == nil then
        return false
      end
  end
  return true
end

local function flatten(input)
  if is_array(input) then
    local result = {}
    for _, value in ipairs(input) do
      for _, elem in ipairs(flatten(value)) do
        table.insert(result, elem)
      end
    end
    return result
  end
  if input then
    return {input}
  end
  return nil
end

return flatten